kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+                        const double reference = r_entity_database.GetValue(time, i_comp, i_gp);
+                        if (!CheckValues(result_array[i_gp][i_comp], reference)) {
+                            FailMessage(it_elem->Id(), "Element", result_array[i_gp][i_comp], reference, p_var_array->Name(), i_comp, i_gp);
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
+        for (auto& p_var_vector : mpGPVariableVectorList) {
+            const auto& r_var_database = r_gp_database.GetVariableData(*p_var_vector);
+
+            #pragma omp parallel for reduction(+:check_counter) firstprivate(result_vector)
+            for (int i = 0; i < static_cast<int>(r_elements_array.size()); ++i) {
+                auto it_elem = it_elem_begin + i;
+
+                const auto& r_entity_database = r_var_database.GetEntityData(i);
+                it_elem->CalculateOnIntegrationPoints(*p_var_vector, result_vector, r_process_info);
+                for (IndexType i_gp = 0; i_gp < result_vector.size(); ++i_gp) {
+                    for (IndexType i_comp = 0; i_comp < result_vector[i_gp].size(); ++i_comp) {
+                        const double reference = r_entity_database.GetValue(time, i_comp, i_gp);
+                        if (!CheckValues(result_vector[i_gp][i_comp], reference)) {
+                            FailMessage(it_elem->Id(), "Element", result_vector[i_gp][i_comp], reference, p_var_vector->Name(), i_comp, i_gp);
+                            check_counter += 1;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    // Final check
+    if (check_counter > 0) {
+        this->Set(CORRECT_RESULT, false);
+    } else {
+        this->Set(CORRECT_RESULT, true);
+    }
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::ExecuteFinalize()
+{
+    KRATOS_TRY;
+
+    // This clears the databse of the result information
+    mDatabaseNodes.Clear();
+    mDatabaseGP.Clear();
+
+    // Set the flag to false
+    this->Set(CORRECT_RESULT, false);
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+int FromJSONCheckResultProcess::Check()
+{
+    KRATOS_TRY;
+
+    const int value_nodes = mDatabaseNodes.Check();
+    const int value_gp = mDatabaseGP.Check();
+
+    return value_nodes + value_gp;
+
+    KRATOS_CATCH("");
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FromJSONCheckResultProcess::InitializeDatabases()
```

Comment:
This method is around 250 lines long. If it is possible, it would be nice to implement some auxiliary methods for the sake of code readability.

Reply:
As a comment aside, I guess that the nodal and elemental loops in here cannot be parallel (just commenting in case you forgot to implement the pragmas).

Reply:
yes, they can be parallel, there are not conflict in the loops